geode-management/src/main/java/org/apache/geode/management/api/ClusterManagementService.java

```diff
@@ -35,7 +35,8 @@
    *        cluster, as well as the group this config belongs to
    * @see CacheElement
    */
-  ClusterManagementResult create(CacheElement config);
+  <T extends CacheElement & RespondsWith<R>, R extends CacheElement> ClusterManagementResult<T> create(
```

Comment:
is `R` necessary here? Should this be simply `<T extends CacheElement> ClusterManagementResult<T> create(
      T config);`

Reply:
That was my first thought as well, but try making that change and it no longer compiles.  All methods, including both `create` and `list` need to call `getConfigurationManager` which itself has both `add` and `list` methods on it, therefore it needs to know about both T and R, therefore all CMS methods need to know about T and R even if they only use T.